pkg/controller/mutate_app_gateway.go

```diff
@@ -59,7 +59,16 @@ func (c AppGwIngressController) GetAppGw() (*n.ApplicationGateway, *appgw.Config
 }
 
 // MutateAppGateway applies App Gateway config.
-func (c AppGwIngressController) MutateAppGateway(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext) error {
+func (c AppGwIngressController) MutateAppGateway(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, invokedForReconciliation bool) error {
```

Comment:
maybe we should do a complete separate MutateAppGateway for the reconcile case.
if somehow client wants to keep any fields of appgw config from the reconcile, the new generated from agic will just wipe them out, hence we need to document what fields agic will not wipe out, e.g. waf policy

Reply:
Fields that are not overriden by AGIC will stay as is.